### PR TITLE
[v1.1 RNN-T] rnnt full dataset optimization

### PR DESCRIPTION
**Pull request by Peach-He**
* add gradient clipping to fix loss NAN issue
*  move layer normalization to config option

### CODE REVIEW

**Comment by xuechendi** [comment]
@Peach-He , this is fixing for RNNT-sda optimized version? What is result WER?

File: modelzoo/rnnt/pytorch/rnnt.patch [hunk 1]
```diff
@@ -3771,15 +3797,16 @@ index 0000000..758849c
 +
 +MODEL_ARGS="--enc_n_hid 1024 \
 +  --enc_pre_rnn_layers 2 \
-+  --enc_stack_time_factor 8 \
++  --enc_stack_time_factor 2 \
```
Comment (xuechendi) [inline modelzoo/rnnt/pytorch/rnnt.patch:3800]: please explain this change

File: modelzoo/rnnt/pytorch/rnnt.patch [hunk 2]
```diff
@@ -4596,42 +4625,20 @@ index de45eb0..8bb5542 100644
  
 -        if args.enable_prefetch:
 -            train_loader.data_iterator().prefetch()
- 
+-
 -        step_start_time = time.time()
-+            if args.enable_prefetch:
```
Comment (xuechendi) [inline modelzoo/rnnt/pytorch/rnnt.patch:4601]: @Peach-He , what is part codes removing for?

File: modelzoo/rnnt/pytorch/rnnt.patch [hunk 3]
```diff
@@ -4749,14 +4767,28 @@ index de45eb0..8bb5542 100644
...
++
++
 +                loss_item= train_step( model, loss_fn, args, batch_size, feats, feat_lens, txt, txt_lens,
 +                                        meta_data, train_loader, rnnt_graph)
 +                step_utts += txt_lens.size(0) * world_size
 +                epoch_utts += txt_lens.size(0) * world_size
 +                accumulated_batches += 1
 +                if accumulated_batches % args.grad_accumulation_steps == 0:
-+
++                    torch.nn.utils.clip_grad_norm_(model.parameters(), args.clip_norm)
```
Comment (xuechendi) [inline modelzoo/rnnt/pytorch/rnnt.patch:4791]: args.clip_norm is passed through config or cmdline?

**Review by xuechendi — APPROVED** [review]
Codes are looking good to me, will merge after adding explanation for reviewer's question

**Comment by Peach-He** [comment]
> @Peach-He , this is fixing for RNNT-sda optimized version? What is result WER?

this PR is optimizations for full dataset to converge to 5.8% WER

File: modelzoo/rnnt/pytorch/rnnt.patch (hunk 1 above)
Comment (Peach-He) [reply modelzoo/rnnt/pytorch/rnnt.patch:3800]: it is model hyper parameter for RNN-T, default `8` result poor converge, change to `2` for default value which can converge to 5.8% WER.

File: modelzoo/rnnt/pytorch/rnnt.patch (hunk 3 above)
Comment (Peach-He) [reply modelzoo/rnnt/pytorch/rnnt.patch:4791]: `args.clip_norm` is a cmdline arg passed in `train.sh`

File: modelzoo/rnnt/pytorch/rnnt.patch (hunk 1 above)
Comment (xuechendi) [reply modelzoo/rnnt/pytorch/rnnt.patch:3800]: Got it, WER as 5.8% just for original RNNT not optimized one right?

**Comment by xuechendi** [comment]
> > @Peach-He , this is fixing for RNNT-sda optimized version? What is result WER?
> 
> this PR is optimizations for full dataset to converge to 5.8% WER

Can you make it clear, so now democratized RNNT reached 5.8% or just stock + optimization RNNT?

File: modelzoo/rnnt/pytorch/rnnt.patch (hunk 1 above)
Comment (Peach-He) [reply modelzoo/rnnt/pytorch/rnnt.patch:3800]: yes, optimized WER is 6.97%

**Comment by Peach-He** [comment]
> > > @Peach-He , this is fixing for RNNT-sda optimized version? What is result WER?
> > 
> > 
> > this PR is optimizations for full dataset to converge to 5.8% WER
> 
> Can you make it clear, so now democratized RNNT reached 5.8% or just stock + optimization RNNT?

latest RNN-T performance data on full dataset: stock model (5.79%), optimized model (6.97%)
